### PR TITLE
Lm 19

### PR DESCRIPTION
**Pull request by BogVin**

### CODE REVIEW

**Review by Chay23 — APPROVED** [review]

File: app/submissions/views.py [hunk 1]
```diff
@@ -1,9 +1,24 @@
...
+from wsgiref.util import FileWrapper
+from django.shortcuts import get_object_or_404
 
 
 class SubmissionViewList(viewsets.ModelViewSet):
     serializer_class = SubmissionSerializer
     queryset = Submission.objects.all()
+
+
+class FileDownloadListAPIView(generics.ListAPIView):
```
Comment (vladhoi) [inline app/submissions/views.py:15]: Why inheriting from ListAPIView? It is used to retrieve a collection of instances. RetrieveAPIView is a better choice here.

File: app/tests/submissions/test_views.py [hunk 2]
```diff
@@ -66,3 +66,25 @@ def test_post_submission_failure(
...
     )
     assert response.status_code == status.HTTP_400_BAD_REQUEST
+
+
+@pytest.mark.django_db
+def test_file_response(user_client, create_valid_submission):
+    """
+    Ensure we can  get  file.
+    """
+    create_valid_submission
```
Comment (vladhoi) [inline app/tests/submissions/test_views.py:76]: Remove this line, valid submission fixture is already in the scope of the test.

File: app/tests/submissions/test_views.py [hunk 3]
```diff
@@ -66,3 +66,25 @@ def test_post_submission_failure(
...
+    response = user_client.get(url)
+    assert response.status_code == status.HTTP_200_OK
+
+
+@pytest.mark.django_db
+def test_file_error_response(create_valid_user, user_client, create_valid_submission):
+    """
+    Ensure we get error 404 while trying to get non-exist file.
+    """
+    create_valid_submission
```
Comment (vladhoi) [inline app/tests/submissions/test_views.py:87]: Remove this line, valid submission fixture is already in the scope of the test.

**Review by vladhoi — CHANGES_REQUESTED** [review]

File: app/assignments/migrations/0001_initial.py [hunk 4]
```diff
@@ -1,24 +0,0 @@
-# Generated by Django 3.1.2 on 2020-12-15 21:02
```
Comment (vladhoi) [inline app/assignments/migrations/0001_initial.py:1]: Do not remove migrations entirely.

**Review by vladhoi — CHANGES_REQUESTED** [review]

**Review by vladhoi — APPROVED** [review]

**Pull request merged**